### ISSUE TITLE
aws_lambda_function data source doesn't fail when function doesn't exist

### ISSUE THREAD

**Issue by apparentlymart** [terraform-providers/terraform-provider-aws#6966]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

```
$ terraform --version
Terraform v0.12.0-dev
+ provider.aws v1.40.0-6-gb23683732-dev
```

I am currently testing with a v0.12.0 prerelease build, but I have verified that this bug is still present in master at the time I write this issue.

### Affected Resource(s)

* `data.aws_lambda_function`

### Terraform Configuration Files

```hcl
data "aws_lambda_function" "auth" {
  function_name = "anything-that-doesnt-exist"
}
```

### Expected Behavior

As with all data sources, the read request should fail with an error if the requested function doesn't exist. 

### Actual Behavior

Because the data source shares the same read function with the corresponding managed resource type, a failure results in an invalid object (with all of its attributes empty) being left in the state.

https://github.com/terraform-providers/terraform-provider-aws/blob/25303d5f4b90adab9d0f08d0d7b397cace7c3532/aws/data_source_aws_lambda_function.go#L151-L154

This is annoying because it causes the problem to be hidden and crop up instead as difficult-to-debug downstream problems, such as errors about required fields on other resources not being populated.

In my case, I was using the `arn` attribute to populate an IAM policy, leading to the following unhelpful error:

```
MalformedPolicyDocument: Syntax errors in policy.
status code: 400, request id: 2b441449-0582-11e9-89f1-b9230e0005a0
```

This is resulting from it attempting to specify the `Resource` in the policy as the empty string, because `data.aws_lambda_function.auth.arn` isn't populated.

**Comment by bflad** [comment]
Fix submitted: #7663 👍

**Comment by bflad** [comment]
The fix for this has been merged and will release in version 2.0.0 of the Terraform AWS Provider, likely in the next week or two. Thanks for reporting this, @apparentlymart!
